dace/libraries/blas/nodes/gemm.py

```diff
@@ -0,0 +1,244 @@
+         c_inputs) = _get_gemm_inputs(node, parent_state, parent_sdfg)
+
+        if c_inputs is not None:
+            edge_c, outer_array_c, shape_c = c_inputs
+
+        if node.transA:
+            trans_shape_a = list(reversed(shape_a))
+        else:
+            trans_shape_a = shape_a
+
+        if node.transB:
+            trans_shape_b = list(reversed(shape_b))
+        else:
+            trans_shape_b = shape_b
+
+        if (len(trans_shape_a) != 2 or len(trans_shape_b) != 2
+                or trans_shape_a[1] != trans_shape_b[0]):
+            raise SyntaxError("Matrix sizes must match")
+        M, K, N = trans_shape_a[0], trans_shape_a[1], trans_shape_b[1]
+        shape_y = (M, N)
+
+        dtype_a = outer_array_a.dtype.type
+        dtype_b = outer_array_b.dtype.type
+        dtype_c = outer_array_c.dtype.type
+        dtype_y = dace.DTYPE_TO_TYPECLASS[np.result_type(dtype_a,
+                                                         dtype_b).type]
+
+        if outer_array_a.storage != outer_array_b.storage:
+            raise ValueError("Input matrices must have same storage")
+        storage = outer_array_a.storage
+
+
+        _, array_a = sdfg.add_array("_a", shape_a, dtype_a, storage=storage)
+        _, array_b = sdfg.add_array("_b", shape_b, dtype_b, storage=storage)
+        _, array_c = sdfg.add_array("_c", shape_c, dtype_c, storage=storage)
+        _, array_y = sdfg.add_array("_y", shape_y, dtype_y, storage=storage)
+
+        if node.alpha == 1.0:
+            mul_program = "__out = __a * __b"
+        else:
+            mul_program = "__out = {} * __a * __b".format(node.alpha)
+
+        if node.beta == 0:
+            mul_out, mul_out_array = "_y", array_y
+            output_nodes = None
+        else:
+            mul_out, mul_out_array = tmp, array_tmp = sdfg.add_temp_transient(
+                shape_y, dtype_y, storage=storage)
+
+            access_tmp = state.add_read(tmp)
+            output_nodes = {mul_out: access_tmp}
+
+
+        # multiplication map
+        state.add_mapped_tasklet(
+            "_MatMult_",
+            {"__i%d" % i: "0:%s" % s
+             for i, s in enumerate([M, N, K])}, {
+                 "__a":
+                 dace.Memlet.simple(
+                     "_a", "__i2, __i0" if node.transA else "__i0, __i2"),
+                 "__b":
+                 dace.Memlet.simple(
+                     "_b", "__i1, __i2" if node.transB else "__i2, __i1")
+             },
+            mul_program, {
+                "__out":
+                dace.Memlet.simple(mul_out,
+                                   "__i0, __i1",
+                                   wcr_str="lambda x, y: x + y",
+                                   wcr_identity=0)
+            },
+            external_edges=True,
+            output_nodes=output_nodes)
+
+        if node.beta != 0:
+            add_program = "__y = ({} * __c) + __tmp".format(node.beta)
+
+            # manually broadcasting C to [M, N]
```

Comment:
It's probably worth adding a `_broadcasted_map` function that does something similar...